src/v/redpanda/request_auth.cc

```diff
@@ -105,8 +105,11 @@ request_auth_result request_authenticator::do_authenticate(
               "Unauthorized", ss::httpd::reply::status_type::unauthorized);
         } else {
             const auto& cred = cred_opt.value();
-            bool is_valid = security::scram_sha256::validate_password(
-              password, cred.stored_key(), cred.salt(), cred.iterations());
+            bool is_valid = (
+              security::scram_sha256::validate_password(
+                password, cred.stored_key(), cred.salt(), cred.iterations())
+              || security::scram_sha512::validate_password(
+                password, cred.stored_key(), cred.salt(), cred.iterations()));
```

Comment:
the assumption with kafka + scram is that the client remembers which shaX variant is used. should we do the same thing: all http admin basic auth uses sha512? or are you trying to provide interop between kafka users and http users? I guess that's a separate discussion: it seems like some orgs would have >> kafka users than http users (e.g. admin users). we could we create a new user type to partition the set of users. just a thought.

Reply:
> the assumption with kafka + scram is that the client remembers which shaX variant is used. should we do the same thing: all http admin basic auth uses sha512? or are you trying to provide interop between kafka users and http users? 

The latter -- existing cloud clusters already have a (kafka) superuser account, and they should be able to use that account to authenticate with the admin API too.

We need to have that broader conversation about account management & account types at some point.  A really general definition of users would decouple the account identity from the auth mechanisms (i.e. a user might have a credential for Kafka and a different credential (or none) for the admin API).